cymetric/convenient_interface.py

```diff
@@ -0,0 +1,584 @@
+    """
+    ref_time.rename(index=str, columns={'TimeStep': 'Time'}, inplace=True)
+
+    if 'SimId' in ref_time.columns.values:
+        ref_time.drop('SimId', 1, inplace=True)
+    df = pd.merge(ref_time, df, how="outer")
+    df.fillna(0, inplace=True)
+    return df
+
+
+def merge_n_drop(df, base_col, add_df, add_col):
+    """
+    Merge some additionnal columns fram an additionnal Pandas Data Frame
+    onother one and then remove the second base column (keeping SimID
+    information).
+    Parameters
+    ----------
+    df : Pandas Data Frame
+    base_col : list of the base columns names
+    add_df : Pandas Data Frame to add in the df one
+    add_col : columns tobe added
+    """
+    df = pd.merge(add_df[add_col], df, on=base_col)
+    df.drop(base_col[1], 1)
+    return df
+
+
+def get_reduced_df(df, rdc_list):
+    """
+    Filter the df Pandas Data Frame according to the rdc_list (list of item
+    in the corresponding columns).
+    Parameters
+    ----------
+    df : Pandas Data Frame
+    rdc_list : list of pair of string and string list.
+    """
+    for rdc in rdc_list:
+        if len(rdc[1]) != 0:
+            df = df[df[rdc[0]].isin(rdc[1])]
+        else:
+            wng_msg = "Empty list provided for " + rdc[0] + " key."
+            warnings.warn(wng_msg, UserWarning)
+    return df
+
+
+def get_transaction_df(evaler_, send_list=[], rec_list=[], commod_list=[]):
+    """
+    Filter the Transaction Data Frame on specific sending facility and
+    receving facility.
+
+    Parameters
+    ----------
+    evaler_ : evaler
+    send_list : list of the sending facility
+    rec_list : list of the receiving facility
+    commod_list : list of the commodity exchanged
+    """
+
+    # initiate evaluation
+    trans = evaler_.eval('Transactions')
+    agents = evaler_.eval('AgentEntry')
+
+    rec_agent = agents.rename(index=str, columns={'AgentId': 'ReceiverId'})
+    if len(rec_list) != 0:
+        rec_agent = rec_agent[rec_agent['Prototype'].isin(rec_list)]
+
+    send_agent = agents.rename(index=str, columns={'AgentId': 'SenderId'})
+    if len(send_list) != 0:
+        send_agent = send_agent[send_agent['Prototype'].isin(send_list)]
+
+    # check if sender and receiver exist
+    if rec_agent.empty or send_agent.empty:
+        return None
+    else:
+        # Clean Transation PDF
+        rdc_list = []
+        rdc_list.append(['ReceiverId', rec_agent['ReceiverId'].tolist()])
+        rdc_list.append(['SenderId', send_agent['SenderId'].tolist()])
+        if len(commod_list) != 0:
+            rdc_list.append(['Commodity', commod_list])
+
+        trans = get_reduced_df(trans, rdc_list)
+
+        # Merge Sender to Transaction PDF
+        base_col = ['SimId', 'SenderId']
+        added_col = base_col + ['Prototype']
+        trans = merge_n_drop(trans, base_col, send_agent, added_col)
+        trans = trans.rename(index=str, columns={'Prototype': 'SenderProto'})
```

Comment:
Perhaps it would be better to spell out everything fully (`SenderPrototype` and `ReceiverProtoype` on line 127 below)?